src/client/src/rt-components/Environment/EnvironmentContext.tsx

```diff
@@ -1,19 +1,39 @@
 import React from 'react'
 
-export interface EnvironmentValue {
-  isDesktop: Boolean
-  openfin?: Window
+export interface EnvironmentValue<Provider extends Window = Window> {
+  provider: Provider
+  openfin?: Provider | null
+  [key: string]: Provider | null
 }
 
 export interface Window {
```

Comment:
I think there is already a global Window type. We should rename this to avoid a conflict.

Reply:
Also pull this to the top of this file.

Reply:
When requesting a rename it's good to offer an alternative — in this case I'm open to suggestions. In this case all I can imagine is `WindowProvider`

Reply:
Regarding order: as the subject of the file is the environment the order is by hierarchy of importance to the reader. If it were `WindowContext` than `Window` would be at the top.